bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZPrimeToMuMu_Allanach/B3mL2/ZPrimeToMuMuSB_Allanach_B3mL2_gzpfit_0p05_thetasb_0p1_run_card.dat

```diff
@@ -0,0 +1,197 @@
+#                                         3=photon from electron     *
+#*********************************************************************
+        1     = lpp1    ! beam 1 type 
+        1     = lpp2    ! beam 2 type
+     6500     = ebeam1  ! beam 1 total energy in GeV
+     6500     = ebeam2  ! beam 2 total energy in GeV
+#*********************************************************************
+# PDF CHOICE: this automatically fixes also alpha_s and its evol.    *
+#*********************************************************************
+     lhapdf    = pdlabel     ! PDF set                                     
+     $DEFAULT_PDF_SETS = lhaid     ! if pdlabel=lhapdf, this is the lhapdf number
+     $DEFAULT_PDF_MEMBERS  = reweight_PDF     ! reweight to get PDF uncertainty
+#*********************************************************************
+# Renormalization and factorization scales                           *
+#*********************************************************************
+ False = fixed_ren_scale  ! if .true. use fixed ren scale
+ False        = fixed_fac_scale  ! if .true. use fixed fac scale
+ 91.188  = scale            ! fixed ren scale
+ 91.188  = dsqrt_q2fact1    ! fixed fact scale for pdf1
+ 91.188  = dsqrt_q2fact2    ! fixed fact scale for pdf2
+ -1 = dynamical_scale_choice ! Choose one of the preselected dynamical choices
+ 1.0  = scalefact        ! scale factor for event-by-event scales
+#*********************************************************************
+# Matching - Warning! ickkw > 1 is still beta
+#*********************************************************************
+ 0        = ickkw            ! 0 no matching, 1 MLM, 2 CKKW matching
+ 1        = highestmult      ! for ickkw=2, highest mult group
+ 1        = ktscheme         ! for ickkw=1, 1 Durham kT, 2 Pythia pTE
+ 1        = alpsfact         ! scale factor for QCD emission vx
+ F        = chcluster        ! cluster only according to channel diag
+ F        = pdfwgt           ! for ickkw=1, perform pdf reweighting
+ 5        = asrwgtflavor     ! highest quark flavor for a_s reweight
+ T        = clusinfo         ! include clustering tag in output
+ 3.0      = lhe_version       ! Change the way clustering information pass to shower.        
+#*********************************************************************
+#**********************************************************
+#
+#**********************************************************
+# Automatic ptj and mjj cuts if xqcut > 0
+# (turn off for VBF and single top processes)
+#**********************************************************
+   T  = auto_ptj_mjj  ! Automatic setting of ptj and mjj
+#**********************************************************
+#                                                                    
+#**********************************
+# BW cutoff (M+/-bwcutoff*Gamma)
+#**********************************
+  15000  = bwcutoff      ! (M+/-bwcutoff*Gamma)
```

Comment:
We are already in contact with the author regarding how these samples should be generated. These are narrow-width samples with a width <<1% of the mass. A small bw cut off is undesirable considering the resolution; hence this arbitrarily large value.

Reply:
@usarica Which author are you talking about? `bwcutoff=15000` is not recommended by the authors of madgraph as this will also create issues for parton shower. Not totally agreeable if author you are referring to is the author of UFO model, author of the external tool (this case, UFO) cannot override the author of the tool itself (this case, madgraph).

Reply:
We have checked the effect of the choice. We summarize our conclusions on [https://1drv.ms/b/s!AhnV79itR6FBjbFalDaM4H7zrnRmow?e=DtLhMF](these slides). We have not observed significant differences in other observables relevant to the analysis (EXO-22-016).

Reply:
@menglu21 @sihyunjeon (Tagging for the response above explicitly)

Reply:
UFO model (author) is an external tool that gets imported by the tool and tool provider themselves are sure of the bug if the value is too large. and bwcutoff is not a setting in UFO model as you'll know. I personally don't think wishing to keep acceptance level high with buggy bsm signal process is a proper claim but i'll leave it up to @menglu21 

But there are other ways that one can generate this with larger spectrum i would assume (also dangerous but conditionally ok)
`generate g g > Zp > mu+ mu- b b~`

Reply:
Hi, here are my 2 cents for what they are worth.

(1) The top left figure in the linked slides from Ulascan show that the LHE (truth) mass distribution with a 15-Gamma cutoff is unphysical, which is no surprise.
(2) The mass distribution with 15K-Gamma cutoff agrees perfectly in both shape and normalization with the 15-Gamma one, within the cutoff region of the latter.  What this tells me is that whatever problem may exist in MadGraph by setting the cutoff at 15K does not affect the peak region.
(3) The mass distribution with the 15K-Gamma cutoff away from the peak region looks more or less OK by eye.  There is nothing obviously pathological going on.  It is  hard to be quantitative since the Breit-Wigner is convoluted with the PDFs.

We then have a choice between a clearly un-phyisical setting (15-Gamma cutoff) and a setting (15K-Gamma) that you claim may be buggy, but if it is buggy the bug only affects the tail.  In practice the differences between the two settings for the use-case of the analysis (a search for a narrow peak!) has been demonstrated to be negligible by pushing the events privately generated with the two settings through the full chain of hadronization, simulation, reconstruction, and analysis.  

At the end of the day it does not make a difference to the analysis whether we use 15-Gamma or 15K-Gamma, so in principle I do not care.  However, we have generated privately a full set of MC events with 15K-Gamma and we are fast approaching pre-approval.   The goal of the exercise is to validate our **full** private MC production chain (GEN+SIM+RECO) by comparing it with the  "official" one for a handful of mass and coupling settings.  To this end I would prefer if you would accept the 15K-Gamma setting, as it would make the comparison of results obtained with the private and official sample a little bit cleaner.

Thank you

Claudio Campagnari, UCSB

Reply:
Hi @usarica sorry for the late reply. To my understanding, the decay chain syntax is valid in a region close to mass peak, where the interference could be ignored. If the interference doesn't exist or you don't care, then it seems ok to me. Let me confirm with MG author, @sihyunjeon can comment also. but the syntax Sihyun suggested could be a workaround since it include both onshell and offshell contribution, but it could be dangerous on the gauge invariant. Regards to the difference during parton shower, i don't have much idea how large it could be, i'll have a check and come back to you.

Reply:
i just realized that i already asked the MG author about it, you can see the effect during parton shower, https://answers.launchpad.net/mg5amcnlo/+question/702105

Reply:
> Hi @usarica sorry for the late reply. To my understanding, the decay chain syntax is valid in a region close to mass peak, where the interference could be ignored. If the interference doesn't exist or you don't care, then it seems ok to me. Let me confirm with MG author, @sihyunjeon can comment also. but the syntax Sihyun suggested could be a workaround since it include both onshell and offshell contribution, but it could be dangerous on the gauge invariant. Regards to the difference during parton shower, i don't have much idea how large it could be, i'll have a check and come back to you.

Dear @menglu21,

Thanks for your response and also for the pointer to the MG discussion. Indeed, we do not consider interference effects with other processes since these resonances still have a relatively small width (interference only matters if the width is large), and particularly in the case of Z'sb couplings, there can in fact be none in the SM at LO in QCD.

The suggestion from Si Hyun to change the process string would not avoid the cutoff value since it would only give rise to the the gg->Z'+bb~, Z'->mu-mu+ diagram, which is already included and observed to be cut.

In order to also check the parton shower point MG authors point out, we added a slide to the comparisons, where we show that the gen.-level b jet pT spectrum does not change at all (note that it also includes b jet multiplicity), so for this particular case, we do not think there should be any issues with parton shower.

The link to the updated slides is the same: https://1drv.ms/b/s!AhnV79itR6FBjbFalDaM4H7zrnRmow?e=DtLhMF

Regards,

Ulascan for the authors

Reply:
@usarica Just for my lesson, can you clarify "The suggestion from Si Hyun to change the process string would not avoid the cutoff value since it would only give rise to the the gg->Z'+bb~, Z'->mu-mu+ diagram, which is already included and observed to be cut." what this means? Are you talking about the other processes that you add (e.g. `g r > Zp j, Zp > mu+ mu-`) ?

Reply:
@sihyunjeon You suggested to write
generate g g > Zp > mu- mu+ b b~
From the generated model, this syntax is equivalent to what we have as
generate g g > Zp b b~, Zp > mu+ mu-
at LO in BSM couplings.

Reply:
No they are different, technically in MadGraph. You end up having same Feynman diagram but whether contributions outside the mass window is included or not should be different. What you did forces onshell Z' (by using `,` in the syntax) and it gives you a restriction in the phase space (mass pole +- bwcutoff X width). If you instead connect the decays through `>` without using `,`, it forces Z' to be included in the Feynman diagram as one of the mediators for the s-channel Feynman diagram (which doesn't necessarily require it to be onshell).

Reply:
Dear @sihyunjeon, @menglu21,

Thanks again for the suggestions for checks, and pointers to the MG discussion. We have updated the slides once more (https://1drv.ms/b/s!AhnV79itR6FBjbFalDaM4H7zrnRmow?e=DtLhMF), adding a comparison of the syntax suggested by Si Hyun, keeping bwcutoff=15. As shown on slide 3, we find that the contributing diagrams, cross section, dimuon invariant mass distribution, and the b jet pT spectrum after Pythia hadronization are all identical to the result of the generation with the parameter we have in this PR. While we do find that the intermediate Z’ is not written to the LHE in ~2% of the cases, this rate is consistent with the proportion of the events away from the Z’ pole mass by more than 15 times the Z’ width. Therefore, we conclude that writing the Z’ intermediate or not in this model has no effect on parton shower, which is expected as the Z’ only decays leptonically, where leptons, having no color charge, do not participate in hadronization.

The event content contains the Z’ particle written in all events with the settings in this PR, and we would prefer these particles written. As the settings are otherwise the same, we would prefer to keep these settings unchanged, and suggest to merge this PR without further delay.

Thanks and regards,

Ulascan for the authors

Reply:
Hi, I don't object to moving on with this request but I would suggest @menglu21 not to merge this as this could possibly taken as a reference by other people who doesn't have a clue on the run card settings.

Just a minor correction, whether Z' is written in the LHE or not doesn't only have effect on decay particles of Z'. It changes the shower scale of the whole frame so it does have effect although likely small.

Reply:
Dear @sihyunjeon, @menglu21, @Saptaparna,

We have updated our slides once again (https://1drv.ms/b/s!AhnV79itR6FBjbFalDaM4H7zrnRmow?e=DtLhMF), adding slide 4 that shows the comparison of the PDF scale recorded. We do not see any noticeable difference. We note that for the showering of all partons in the event, this scale is taken as the basis in the default CMS parton shower configuration. For the FSR from muons, we read from the Pythia documentation that timelike showers feature near-NLO accuracy using the full phase space when resonances are involved. Nevertheless, any difference in showering due to not recording the resonance has negligible consequence to our purposes. In the end, the target analysis is a search for a Z', not a precision measurement of its kinematics, for which this MC configuration is not suitable to begin with.

Regarding the point on the sign-off and merging of the PR, we do not think the parameters we set are unprecedented in genproductions. Depending on the needs of analyses, it looks like different choices have already been made. For example, all of these cases feature bwcutoff=15000, in some cases likely because the resonances are narrow: https://github.com/search?p=1&q=org%3Acms-sw+%2215000+%3D+bwcutoff%22&type=Code. Taking one of those cards as an example, https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/cards/production/pre2017/13TeV/AtoZh_mhmodp_MG5_aMCNLO/AtoZh_M340/AtoZh_M340_proc_card.dat, we do see the process syntax to also be the same. Therefore, we do not see the argument to set precedence as valid.

We believe the PR should be merged in order for the cards to be referenced properly in central CMS production, which is what we would like to ultimately have for at least one of the samples for the purpose of validating the private MC. In the interest of keeping this discussion focused, we would like to kindly ask the GEN conveners @menglu21 @Saptaparna to provide their opinion on how validation should proceed as this is our primary goal.

Thanks,

Ulascan for the authors

Reply:
I cannot disagree more on your replies. The link you provided was in 2017 [1], and this is exactly the bad example that people can take it as a reference, so again it should not be merged although production can go on. We even had SM W process run card example in genproductions with bwcutoff with 15000 or 5000 for several years as well which is totally wrong. The ones in genproductions are not updated for the old cards that has been merged already, these merge requests are just for green light purposes and it doesn't give you forever longing definite answers. There are somethings that previous reviewers didn't know possible issues with wrong settings (and again, bwcutoff is the example) or they might've made a mistake although they knew the issue. What we know for sure right now is "although the effect seems to be negligible, what is wrong is wrong and there is clearly better way to model your same process without using `,` in the syntaxes".

The reason why I am saying this would be OK to be produced but should not be merged into genproductions is because
a. There is a better way to do it but you don't want to reproduce your samples so it's a compromise in between. This doesn't mean your samples are bug free.

b. You've checked the relevant variables for your analysis and you found them consistent so that's OK. But again, this doesn't mean your samples are bug free, we don't know for some other observable something might be wonky. But since we cannot possibly check millions of observables with the process, your check seems fine and it seems valid for your analysis specifically.
e.g. With wrong settings of W process, there was a study on comparing the effect of bwcutoff and W pT distribution was different. This might not be for your case (Z' pT might be the same) but we would never know what could be really wrong.

c. We would repeat the same mistakes everytime unless we stop merging such things in (also bad for reviewers since they have to repeat the same answers)


If I may give you an easy example, there are a. best way, b. good or bad but still allowable way, and c. bad way to make W process.

a. p p > e+ ve : where you take on- and off-shell contributions precisely as much as the tool (this case madgraph) advises you to

b-1. p p > w+ > e+ ve : where you take all possible s-channel contributions but what the authors asks to use with caution

b-2. p p > w+, w+ > e+ ve : where you restrict your phase space to onshell W, this does not give you full spectrum of the physics that W process has but can be used if the user knows it's restrictions.

c. p p > e+ ve with large bwcutoff : which is deprecated by the authors themselves already.

Again, your physics modelling can be done with other syntaxes without using `,`. But it seems that for your analysis purposes things look fine. However this still doesn't mean this could be taken as a reference in the future as well since clearly there is a better way to properly do it.

[1] https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/cards/production/pre2017/13TeV/AtoZh_mhmodp_MG5_aMCNLO/AtoZh_M340/AtoZh_M340_proc_card.dat

Reply:
Dear @sihyunjeon, @menglu21, @Saptaparna,

As mentioned in our previous reply, our primary concern is for a validation sample to be produced through central production. If the steps outlined for EXO (https://exo-mc-and-i.gitbook.io/exo-mc-and-interpretation/how-to-sample-request) can be accomplished without merging this PR, we can proceed as such since you mention "this would be OK to be produced". Please kindly clarify the procedure in this case.

Another item in the procedure seems to upload the gridpacks to the central cvmfs location. We provided the gridpacks at the first message posted in this PR, davs://redirector.t2.ucsd.edu:1095//store/user/usarica/ZPrime2022_Gridpacks/models, and also later in /afs/cern.ch/work/u/usarica/public/ZPrime2022_Gridpacks/models as an alternative location. We can choose the Y3 model with M=1 TeV (tarball ZPrimeToMuMuSB_Allanach_Y3_M1000_gzpfit_0p14_thetasb_-0p15_slc7_amd64_gcc700_CMSSW_10_6_19_tarball.tar.xz) for the central production validation sample. Would this work?

Regards,

Ulascan for the authors

Reply:
Hi @usarica Thanks for all your responses. @sihyunjeon thanks for your suggestions @menglu21  and I will take a quick look and get back to you as soon as we can.

Reply:
Hi @usarica I have a quick question regarding the BWCutOff -- do you notice any performance issues. I extended the BWCutOff here: https://answers.launchpad.net/mg5amcnlo/+question/696189 and noticed that fewer and fewer events were being generated as the BWCutOff was set to higher values. 

My suggestion would be to merge the PR with the caveat that this is a very specific case and the set-up should not be replicated for other MC requests. I have not in favor of setting the precedent that we generate samples without merged PRs.

Reply:
Also, VPrime_Allanach_B3mL2_UFO.tar.xz,  VPrime_Allanach_DY3_UFO.tar.xz,  VPrime_Allanach_DYp3_UFO.tar.xz,  VPrime_Allanach_Y3_UFO.tar.xz are already on https://cms-project-generators.web.cern.ch/cms-project-generators/ Is there anything else that should be done, from our(=GEN) end, before sample production can begin?

Reply:
Hi @sihyunjeon Regarding to PR merging, actually i have different opinion on it. I would like to merge it. We know that the suggested bwcutoff is 15 from the author side, with this PR, we can tell people that detailed checks should be if they want some other values, especially when using some extremely large value.

Reply:
@Saptaparna @menglu21 hi,

> My suggestion would be to merge the PR with the caveat that this is a very specific case and the set-up should not be replicated for other MC requests.

We didn't provide any guidelines what is the basis that people should follow (again, half joking and a bit unrelated to this - the tutorial for gridpacks hasn't happened although we are on the verge run3 production :-|). And the following problem is, we don't really have a flag in our genproductions whether something is OK to be copy pasted or not.

> I am not in favor of setting the precedent that we generate samples without merged PRs.

We actually have a lot of samples already as you'll know that goes in without any merge requests but OK this is up to you.

> We know that the suggested bwcutoff is 15 from the author side, with this PR, we can tell people that detailed checks should be if they want some other values, especially when using some extremely large value

There needs to be a bit different wording for this. It is not really "suggested" value, I had an email exchange with olivier 3 years ago and he clearly stated "it is bad and dangerous for both in terms of physics and computing" (computing related to what sapta pointed out to - good catch). So I don't think it's even OK to allow people using sth with detailed checks which is in the first place not allowable from the author of the tools. We never know what is the level of "enoughly detailed checks" - we didn't expect the PS weight uncertainty to break so there is still enough possibility that it could be wonky we are not aware of.

Reply:
@usarica I already told EXO convenors (nonhadronic L3 as well) last week that this is OK for central production and there is no issue for requesting samples centrally from EXO side. Rest is up to GEN (and then later on PdmV) and based on their comments I don't see any stopping factors from the physics point of view.